src/register.ts

```diff
@@ -56,31 +57,68 @@ export async function register(
   avatar: string | undefined,
   about: string,
   externalResources: IExternalResource[],
-  captchaToken: string,
+  captchaToken: string | undefined,
+  captchaBypassKey: string | undefined,
   callback: RegisterCallback
 ) {
   // verify captcha if enabled
   if (HCAPTCHA_ENABLED) {
-    if (!captchaToken) {
+    if (captchaBypassKey && captchaBypassKey !== CAPTCHA_BYPASS_KEY) {
```

Comment:
When I initially looked at it, it seemed correct but after testing I realized some cases are not handled correctly, so if key is correct, it doesn't by pass captcha.

We also need to throttle auth attempts, to prevent brute forcing of key.

You moved the api throttling to the top, coming before all the input validation. This is a bad idea because it can lead to DoS. I should have probably left a comment.